tools/openblas_support.py

```diff
@@ -232,36 +232,13 @@ def make_init(dirname):
     with open(os.path.join(dirname, '_distributor_init.py'), 'wt') as fid:
         fid.write(textwrap.dedent("""
             '''
-            Helper to preload windows dlls to prevent dll not found errors.
-            Once a DLL is preloaded, its namespace is made available to any
-            subsequent DLL. This file originated in the numpy-wheels repo,
-            and is created as part of the scripts that build the wheel.
+            Helper to add the path to the openblas dll to the dll search space
             '''
             import os
-            import glob
-            if os.name == 'nt':
-                # convention for storing / loading the DLL from
-                # numpy/.libs/, if present
-                try:
-                    from ctypes import WinDLL
-                    basedir = os.path.dirname(__file__)
-                except:
-                    pass
-                else:
-                    libs_dir = os.path.abspath(os.path.join(basedir, '.libs'))
-                    DLL_filenames = []
-                    if os.path.isdir(libs_dir):
-                        for filename in glob.glob(os.path.join(libs_dir,
-                                                               '*openblas*dll')):
-                            # NOTE: would it change behavior to load ALL
-                            # DLLs at this path vs. the name restriction?
-                            WinDLL(os.path.abspath(filename))
-                            DLL_filenames.append(filename)
-                    if len(DLL_filenames) > 1:
-                        import warnings
-                        warnings.warn("loaded more than 1 DLL from .libs:"
-                                      "\\n%s" % "\\n".join(DLL_filenames),
-                                      stacklevel=1)
+            import sys
+            extra_dll_dir = os.path.join(os.path.dirname(__file__), '.libs')
+            if sys.platform == 'win32' and os.path.isdir(extra_dll_dir):
+                os.add_dll_directory(extra_dll_dir)
```

Comment:
This may not work for Python <3.10, see https://github.com/conda/conda/issues/10897#issuecomment-977576891. Is it actually needed? If not, I'd really prefer to not touch this thing.

Reply:
FWIW, I think the problem is the opposite: PATH manipulation no longer works in pythons provided by python.org , but still works in conda-provided pythons up to python3.10. `add_dll_directory` did not exist (the code preceeds python3.8), so at the time preloading the dlls was the only way to make this support all the versions of python supported by NumPy. Since we dropped support for <3.7, all versions should have `add_dll_directory`, so this is much cleaner.

I will move it to a separate PR.

Reply:
> PATH manipulation no longer works in pythons provided by python.org

I don't think this is true. The current contents work fine, AFAIK we have not had any trouble with them for a couple of years now. Unless I missed the relevant issue? If there is no issue, I am pretty sure we should not touch this at all until Python 3.10 is the minimum supported version.

Reply:
Ah I see you already opened the other PR.